### PR TITLE
Update voting restrictions

### LINKED ISSUE #144: Sort poll:voted_id
Sort this field in the restrictor, if it is send to users.

### PR DESCRIPTION
**Pull request by jsangmeister**
- auto-calculate `votescast` for not-finished polls
- sort `voted_ids` (fixes #144)

### CODE REVIEW

**Review by ostcar — APPROVED** [review]
I added a review directly in commits. Please have a look.

The last step is to fix the test. This is not easy in os3. We generated tests from the python code to make sure, that the go code creates the same output. This was very good when we wrote the service, but it makes it nearly impossible to change the behavior. 

You have to change the file `internal/test/export.json`. Afterwards, call `go generate ./...` to create the go-files. Then the tests should run again.

File: internal/apps/poll/restrict.go [hunk 1]
```diff
@@ -132,28 +151,51 @@ func RestrictOption(r restricter.HasPermer, canSee, canManage string) restricter
...
+		var state int
+		if err := json.Unmarshal(vote["pollstate"], &state); err != nil {
+			return nil, fmt.Errorf("unmarshal pollstate: %w", err)
+		}
+
+		if state == StatePublished {
+			return element, nil
+		}
+
+		// Th user_token is only visable in the published state. Even a manager
```
Comment (FinnStutzenstein) [inline internal/apps/poll/restrict.go:179]: This sounds not right. The token should be visible in finished and published state.

File: internal/apps/poll/restrict.go [hunk 2]
```diff
@@ -72,8 +76,13 @@ func RestrictPoll(r restricter.HasPermer, canSee, canManage string, restrictedFi
 			return nil, fmt.Errorf("unmarshal poll state: %w", err)
 		}
 
-		if state != StatePublished && state != StateFinished {
```
Comment (FinnStutzenstein) [inline internal/apps/poll/restrict.go:75]: Please stick to this form: Explicitly name the states, which are allowed to see sensitive things.

Comment (jsangmeister) [reply internal/apps/poll/restrict.go:75]: Done

File: internal/apps/poll/restrict.go (hunk 1 above)
Comment (jsangmeister) [reply internal/apps/poll/restrict.go]: Done.

**Comment by jsangmeister** [comment]
Thank you for your input @ostcar, I'm still not very familiar with go. I Implemented the requested changes. Adjusting the tests will probably take more time and will follow probably on thursday.

**Review by FinnStutzenstein — APPROVED** [review]

File: internal/apps/poll/restrict.go [hunk 3]
```diff
@@ -132,28 +151,54 @@ func RestrictOption(r restricter.HasPermer, canSee, canManage string) restricter
...
+		if err := json.Unmarshal(vote["pollstate"], &state); err != nil {
+			return nil, fmt.Errorf("unmarshal pollstate: %w", err)
+		}
+
+		if state == StatePublished {
+			return element, nil
+		}
+
+		// The user_token is only visible in the published state, and finished state
+		// for managers
```
Comment (ostcar) [inline internal/apps/poll/restrict.go:181]: Not important. Only for the future. Could you end a comment with a dot? The only exception is, when the comment is kind of a headline.

**Pull request merged**